cpp/src/io/parquet/page_data.cu

```diff
@@ -92,6 +92,25 @@ struct page_state_s {
   int32_t row_index_lower_bound;              // lower bound of row indices we should process
 };
 
+/**
+ * @brief Returns whether or not a page spans either the beginning or the end of the
+ * specified row bounds
```

Comment:
Does this intentionally cover these 2 cases?
 * A page is not fit in the given boundary
 * A page just sits at the row boundary but it is fit in the boundary

Reply:
I think this logic handles all the cases correctly:

```
     |---|                                                         |---|
     A  B                                                          A   B
            |----------------------------------------------------|
            A                                                    B
                  |PPPPPPPPPPPPPPPPPPPPPPPPPPPPPPPPPPPPPPPPPP|
                  X                 Page                    Y

              |---|                                          |---|
              A   B                                          A   B
              |-------|     |---|                        |------|
              A       B     A   B                        A       B
```
The top cases fail and the bottom cases pass.   We want to include the edge cases where B==X and A==Y because those represent the case where "Page" contains values from either the prev/next row.  Although maybe using "max_row" is a bit deceptive here, since `max_row` would really be `(min_row + num_rows)-1`

Reply:
then maybe begin and end fit better than min and max?